docs/tutorials/spec2006-tutorial-details.md

```diff
@@ -0,0 +1,636 @@
+
+### Building gem5
+In this step, we download the source code and build gem5 v20.1.0.2.
+
+```sh
+git clone -b v20.1.0.2 https://gem5.googlesource.com/public/gem5
+cd gem5
+scons build/X86/gem5.opt -j8
+```
+
+We have two artifacts: one is the gem5 source code (the gem5 git repo), and the gem5 binary (`gem5.opt`).
+The documentation of this step would be how we get the source code and how we compile the gem5 binary.
+In launch_spec2006_experiments.py, we document the step in Artifact objects as follows,
+
+```python
+gem5_repo = Artifact.registerArtifact(
+    command = '''
+        git clone -b v20.1.0.2 https://gem5.googlesource.com/public/gem5
+        cd gem5
+        scons build/X86/gem5.opt -j8
+    ''',
+    typ = 'git repo',
+    name = 'gem5',
+    path =  'gem5/',
+    cwd = './',
+    documentation = 'cloned gem5 v20.1.0.2'
+)
+
+
+gem5_binary = Artifact.registerArtifact(
+    command = 'scons build/X86/gem5.opt -j8',
+    typ = 'gem5 binary',
+    name = 'gem5-20.1.0.2',
+    cwd = 'gem5/',
+    path =  'gem5/build/X86/gem5.opt',
+    inputs = [gem5_repo,],
+    documentation = 'compiled gem5 v20.1.0.2 binary'
+)
+```
+
+### Building m5
+m5 is a binary that facilitates the communication between the host system and the guest system (gem5).
+The examples of how m5 is used could be found in the runscripts that we will describe later.
+m5 binary will be copied to the disk image, so that the guest could run m5 binary during the experiment.
+Therefore, m5 binary should be compiled before we build the disk image.
+
+**Note:** it's important to compile the m5 binary with -DM5_ADDR=0xFFFF0000 as is default in the SConscript.
+If you don't compile with -DM5_ADDR and try to run with KVM, you'll get an illegal instruction error.
+
+To compile m5 binary,
+
+```sh
+cd gem5/util/m5/
+scons build/x86/out/m5
+```
+
+In launch_spec2006_experiments.py, we document the step in an Artifact object as follows,
+
+```python
+m5_binary = Artifact.registerArtifact(
+    command = 'scons build/x86/out/m5',
+    typ = 'binary',
+    name = 'm5',
+    path =  'gem5/util/m5/build/x86/out/m5',
+    cwd = 'gem5/util/m5',
+    inputs = [gem5_repo,],
+    documentation = 'm5 utility'
+)
+```
+
+### Preparing Scripts to Modify the Disk Image
+In this step, we will prepare the scripts that will modify the disk image after the Ubuntu installation process has finished, and before the first time we use the disk image in gem5.
+We will keep the related files in the disk-image folder of the experiment.
+The files that are made specifically for SPEC 2006 benchmarks will be in `disk-image/spec-2006`, and the files that are commonly used accross most benchmarks will be in `disk-image/shared`.
+
+In the root folder of the experiment,
+
+```sh
+mkdir disk-image
+mkdir disk-image/spec-2006
+mkdir disk-image/shared
+```
+
+The first script is the `runscript.sh` script, which will be appended to the end of `.bashrc` file.
```

Comment:
I would explain the `.bashrc` file before leading into this. It may confuse people. You're not talking about the host's .bashrc file.

Reply:
clarified the .bashrc file